multimedia/ffmpeg7/files/patch-libavdevice-avfoundation.diff

```diff
@@ -0,0 +1,14 @@
+--- libavdevice/avfoundation.m
++++ libavdevice/avfoundation.m
+@@ -763,6 +763,11 @@ static int get_audio_config(AVFormatContext *s)
+     return 0;
+ }
+
++#if (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE && __IPHONE_OS_VERSION_MAX_ALLOWED < 110000) || \
++    ((!defined(TARGET_OS_OSX) || TARGET_OS_OSX) && MAC_OS_X_VERSION_MAX_ALLOWED < 101300)
++typedef NSString* AVMediaType;
++#endif
++
+ static NSArray* getDevicesWithMediaType(AVMediaType mediaType) {
+ #if ((TARGET_OS_IPHONE && __IPHONE_OS_VERSION_MIN_REQUIRED >= 100000) || (TARGET_OS_OSX && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101500))
```

Comment:
Here the source uses underscore-prefixed versions. If the file uses AvailabilityMacros.h and not Availability.h, this is wrong. If it uses Availability.h, then perhaps use the same style in the patched fragment too.

Reply:
The iOS macros do not have counterparts without underscores

Reply:
I only refer to MacOS macros here. (Away from laptop atm, hard to check which headers are used.)

Reply:
I have used the macros without underscores in my code. The existing code does use the ones with underscores, but there are 19 occurences all over the file already. The patch would become quite messy if I were to replace all of them. Unrelated changes in later versions would also have a higher chance of breaking it.

Reply:
It is not clear what even includes _any_ `Availability*.h`. Possible Apple SDK from AVFoundation, but then it has to be verified.

FFMpeg itself seems not to include either header in any relevant to the case places.

Reply:
`AVFoundation.h` includes `AVBase.h`, which in turn includes `Availability.h`. Also, `AvailabilityMacros.h` must have been included indirectly, because that is where the underscoreless macros are defined, but the exact chain is not clear to me (a quick `grep` did not give me anything obvious).

Reply:
If `Availability.h` is included, then perhaps rather use underscore-prefixed macros to match the code style at least.